packages/core/src/lib/implementation/read-rc-file.ts

```diff
@@ -0,0 +1,51 @@
+import { join } from 'path';
+import {
+  CONFIG_FILE_NAME,
+  CoreConfig,
+  coreConfigSchema,
+} from '@code-pushup/models';
+import { fileExists, importEsmModule } from '@code-pushup/utils';
+
+export class ConfigPathError extends Error {
+  constructor(configPath: string) {
+    super(`Provided path '${configPath}' is not valid.`);
+  }
+}
+
+export async function readRcByPath(filepath: string): Promise<CoreConfig> {
+  if (!filepath.length) {
+    throw new Error('The path to the configuration file is empty.');
+  }
+
+  if (!(await fileExists(filepath))) {
+    throw new ConfigPathError(filepath);
+  }
+
+  return importEsmModule<CoreConfig>(
+    {
+      filepath,
+    },
+    coreConfigSchema.parse,
+  );
+}
+
+export async function readRc(): Promise<CoreConfig> {
```

Comment:
The name of the function is misleading. A better name would be `searchForRcConfig` or similar, that indicates that the function is not simply reading a file but it is scanning the root folder to find one of three expected files.

Reply:
named it `autoloadRc`